### PR TITLE
feat(go): implement load prompt from source

### PR DESCRIPTION
**Pull request by josearomeroj**
Usage:
```go
	source := `---
description: Test Prompt
model: test/model
input:
  schema:
    username: string
---
Hello {{username}}`

	p := LoadPromptFromSource(g, source, "testPrompt", "testNamespace")
```

Checklist (if applicable):
- [x] PR title is following https://www.conventionalcommits.org/en/v1.0.0/
- [x] Tested (manually, unit tested, etc.)
- [ ] Docs updated (updated docs or a docs bug required)

### CODE REVIEW

File: go/ai/prompt_test.go [hunk 1]
```diff
@@ -1506,3 +1506,52 @@ func TestWithOutputSchemaName_DefinePrompt_Missing(t *testing.T) {
...
+	if len(req.Messages) == 0 || len(req.Messages[0].Content) == 0 {
+		t.Fatal("Render produced empty messages")
+	}
+
+	gotText := req.Messages[0].Content[0].Text
+	wantText := "Hello Developer"
+	if gotText != wantText {
+		t.Errorf("Render got text %q, want %q", gotText, wantText)
+	}
+}
```
[the remark this answers is not in the document]
Comment (josearomeroj) [reply go/ai/prompt_test.go:1557]: We cannot add a test case that expects LoadPromptFromSource to return nil for invalid template syntax (like an unclosed variable) because the underlying dotprompt library performs parsing loosely at load time. It does not validate the template syntax during Parse, so LoadPromptFromSource successfully returns a Prompt object. The error would only occur later during execution (e.g., when Render is called).

**Comment by apascal07** [comment]
Can I ask what your use case for this is? Why not define a prompt programmatically with strong types, etc?

**Comment by josearomeroj** [comment]
> Can I ask what your use case for this is? Why not define a prompt programmatically with strong types, etc?

I need this to treat prompts as dynamic data by loading them from an external databas at runtime, also enables the use of go:embed strings to register prompts directly from embedded files.

**Comment by apascal07** [comment]
> > Can I ask what your use case for this is? Why not define a prompt programmatically with strong types, etc?
> 
> I need this to treat prompts as dynamic data by loading them from an external databas at runtime, also enables the use of go:embed strings to register prompts directly from embedded files.

Got it. FYI, we are adding embed support: https://github.com/firebase/genkit/pull/3973

Let me consider this, we'll either get this PR in or something else that works for you by the next release. Thanks for your contribution!

**Comment by apascal07** [comment]
Added support for it here: https://github.com/firebase/genkit/pull/3973
